asyncapi/v2/v2.go

```diff
@@ -297,8 +327,21 @@ func (o Operation) IsClientSubscribing() bool {
 
 func (o Operation) Messages() []asyncapi.Message {
 	if o.MessageField != nil {
-		return []asyncapi.Message{o.MessageField}
+		if len(o.MessageField.Payload().OneOf()) == 0 {
+			return []asyncapi.Message{o.MessageField}
+		}
+
+		var msgs []asyncapi.Message
+		for _, payload := range o.MessageField.Payload().OneOf() {
+			p := payload.(*Schema)
+			msgs = append(msgs, Message{
+				PayloadField: p,
+			})
+		}
+
+		return msgs
```

Comment:
It's not the true. `MessageField` should have `OneOf` property and you should use it, not `Payload.OneOf()`. `Payload.OneOF` is related to single message, not to array of messages :)

Reply:
Wow, this is not explained at all in the documentation :O Well, only through an example but not clear at all: https://www.asyncapi.com/docs/specifications/v2.0.0#messageObject

Is there any other exception or hidden gem that you know about that I should add here?

Reply:
You have information about `oneOf` in the https://www.asyncapi.com/docs/specifications/v2.0.0#operationObject

for the message field, you can read:

> A definition of the message that will be published or received on this channel. oneOf is allowed here to specify multiple messages, however, a message MUST be valid only against one of the referenced message objects.

so you can have shape:

```yaml
publish:
  message
    payload: ...
```

or

```yaml
publish:
  message:
    oneOf:
      - payload: 
      - payload: 
```

Reply:
Yeah, This `oneOf` field should be either part of the message object or a new type that holds that field.
Ok, gotta work on it! thks!

Reply:
This is done

Reply:
For reference https://github.com/asyncapi/spec/issues/588